src/vs/workbench/contrib/accessibility/browser/accessibleView.ts

```diff
@@ -0,0 +1,160 @@
+		const codeEditorWidgetOptions: ICodeEditorWidgetOptions = {
+			contributions: EditorExtensionsRegistry.getSomeEditorContributions([LinkDetector.ID, SelectionClipboardContributionID, 'editor.contrib.selectionAnchorController'])
+		};
+		const editorOptions: IEditorConstructionOptions = {
+			...getSimpleEditorOptions(),
+			lineDecorationsWidth: 6,
+			dragAndDrop: true,
+			cursorWidth: 1,
+			wrappingStrategy: 'advanced',
+			wrappingIndent: 'none',
+			padding: { top: 2, bottom: 2 },
+			quickSuggestions: false,
+			renderWhitespace: 'none',
+			dropIntoEditor: { enabled: true },
+			accessibilitySupport: this._configurationService.getValue<'auto' | 'off' | 'on'>('editor.accessibilitySupport'),
+			cursorBlinking: this._configurationService.getValue('terminal.integrated.cursorBlinking'),
+			readOnly: true
+		};
+		this._editorWidget = this._register(this._instantiationService.createInstance(CodeEditorWidget, this._editorContainer, editorOptions, codeEditorWidgetOptions));
+	}
+
+	registerProvider(provider: IAccessibleContentProvider): void {
+		this._providers.set(provider.id, provider);
+	}
+
+	show(providerId: string): void {
+		const delegate: IContextViewDelegate = {
+			getAnchor: () => this._editorContainer,
+			render: (container) => {
+				return this._render(providerId, container);
+			},
+			onHide: () => {
+				this._providers.get(providerId)?.onClose();
+			}
+		};
+		this._contextViewService.showContextView(delegate);
+	}
+
+	private _getProviderOrThrow(providerId: string): IAccessibleContentProvider {
+		const provider = this._providers.get(providerId);
+		if (!provider) {
+			throw new Error(`No accessible view provider with id: ${providerId}`);
+		}
+		return provider;
+	}
+
+	private _render(providerId: string, container: HTMLElement): IDisposable {
+
+		const provider = this._getProviderOrThrow(providerId);
+
+		let fragment = '';
+		if (provider.options?.isHelpMenu) {
+			fragment += localize('introMsg', "Welcome to {0} Accessibility Help. Exit this menu and return to the {0} via the Escape key.\n", providerId);
+		}
+		fragment += provider.provideContent();
+
+		this._getTextModel(URI.from({ path: `accessible-view-${providerId}`, scheme: 'accessible-view', fragment })).then((model) => {
+			if (!model) {
+				return;
+			}
+			this._editorWidget.setModel(model);
+			const domNode = this._editorWidget.getDomNode();
+			if (!domNode) {
+				return;
+			}
+			container.appendChild(domNode);
+			this._layout();
+			this._register(this._editorWidget.onKeyDown((e) => {
+				if (e.keyCode === KeyCode.Escape) {
+					this._contextViewService.hideContextView();
+				}
+			}));
+			this._register(this._editorWidget.onDidBlurEditorText(() => this._contextViewService.hideContextView()));
+			const label = provider.options?.isHelpMenu ? 'accessibility help' : 'accessible view';
+			this._editorWidget.updateOptions({ ariaLabel: localize('accessible-view-label', "{0} {1}", providerId, label) });
```

Comment:
I wouldn't expect an id to show up in a string exposed to the user. Also in some languages the id in this sentence could appear after accessibility help. Just passing in the full label via options would be simpler.